library/std/src/io/mod.rs

```diff
@@ -2389,16 +2389,50 @@ impl<T: Read, U: Read> Read for Chain<T, U> {
         }
         self.second.read_vectored(bufs)
     }
+
+    #[inline]
+    fn is_read_vectored(&self) -> bool {
+        self.first.is_read_vectored() || self.second.is_read_vectored()
+    }
```

Comment:
Shouldn't this require both to be efficient? I could be persuaded otherwise, certainly, but can you explain why you chose this logic instead? If one half of the chain doesn't have an efficient implementation, then why is the chained type efficient?

```suggestion
    #[inline]
    fn is_read_vectored(&self) -> bool {
        self.first.is_read_vectored() & self.second.is_read_vectored()
    }
```

Reply:
I guess you could also use a more complicated answer which checks `done_first`, but that seems prone to confusion, even if it might be "more accurate".